sql/src/main/java/io/crate/planner/operators/HashJoin.java

```diff
@@ -172,7 +173,7 @@ public ExecutionPlan build(PlannerContext plannerContext,
             leftHashInputs,
             rightHashInputs,
             Symbols.typeView(lhs.outputs()),
-            lhs.numExpectedRows());
+            Paging.PAGE_SIZE);
```

Comment:
what if the page size was updated and adjusted?

Reply:
this just a first step. adjustable page size and HEAP based math are different WPs